utils/imgutils/gif.go

```diff
@@ -274,26 +273,22 @@ func (d *decoder) readHeaderAndScreenDescriptor() error {
 	if fields := d.tmp[10]; fields&fColorTable != 0 {
 		d.backgroundIndex = d.tmp[11]
 		// readColorTable overwrites the contents of d.tmp, but that's OK.
-		if d.globalColorTable, err = d.readColorTable(fields); err != nil {
+		if err = d.readColorTable(fields); err != nil {
 			return err
 		}
+		d.hasGlobalColorTable = true
 	}
 	// d.tmp[12] is the Pixel Aspect Ratio, which is ignored.
 	return nil
 }
 
-func (d *decoder) readColorTable(fields byte) (color.Palette, error) {
+func (d *decoder) readColorTable(fields byte) error {
```

Comment:
Nit: if we don't read the color table anymore, could we rename this to something like `validateColorTable`?

Reply:
Technically we are reading the table but we are not storing it anywhere. I think the name is correct as there's isn't any special validation there but purely checking that data is there.

Reply:
Sounds good :)

> purely checking that data is there

I'd say that's a validation, even if it's just the bare minimum :stuck_out_tongue:

Reply:
We are validating the file as a whole but not specifically the color table though. I think the method is doing what it says which is reading the data matching the color table. Anyway, this is semantics at this point :p